### DISCUSSION TITLE
cara merubah 1k menjadi 1000 di r

### DISCUSSION THREAD

**Discussion by fatwa13** [teamalgoritma/community#212] [Q&A] [answered]
selamat siang bapak/ ibu , saya mau bertanya untuk cara merubah 1k menjadi 1000 bagaimana ya ? saya coba di stackoverflow  tidak bekerja terima kasih

![algoritma](https://user-images.githubusercontent.com/59698765/179668116-cc91ab6c-3d3f-40e7-80dc-b2d9f66955a3.png)

**Comment by idoali** [comment, marked as answer]
Dalam kasus ini, bapak bisa membuat fungsi baru sebagai berikut : 
```{R}
return.new.numbers <- function(inp){
  y <- ifelse(str_extract_all(inp, '\\D') == "k", 
              as.numeric(str_extract_all(inp, '\\d.')) * 1000, 
              as.numeric(str_extract_all(inp, '\\d.')) * 1000000)
  return(y)
}
```
Fungsi ini berguna untuk mengalikan data dengan angka 1000 semisal data diakhiri oleh huruf `K`, atau mengalikannya dengan angka 1e6 jika diakhiri huruf yang lain. Fungsi ini nanti bisa langsung diaplikasikan, salah satunya menggunakan `mutate()`. Jika ingin dilanjutkan dengan `mutate()` maka :
```{R}
data_tiktok <- data_tiktok %>% 
  mutate(Subscribers.count=return.new.numbers(Subscribers.count))
```

**Comment by fatwa13** [comment]
ok terima kasih , alhamdulillah sudah bisa
